server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java

```diff
@@ -220,15 +221,16 @@ public static List<Pair<Boolean, TimeValuePair>> calculateLastPairForSeriesLocal
         resultContainer.get(i).right = readerList.get(index++).readLastPoint();
         if (resultContainer.get(i).right.getValue() != null) {
           resultContainer.get(i).left = true;
-          if (CACHE_ENABLED) {
-            cacheAccessors.get(i).write(resultContainer.get(i).right);
-            if (context.isDebug()) {
-              DEBUG_LOGGER.info(
-                  "[LastQueryExecutor] Update last cache for path: {} with timestamp: {}",
-                  seriesPaths,
-                  resultContainer.get(i).right.getTimestamp());
-            }
+          cacheAccessors.get(i).write(resultContainer.get(i).right);
+          if (context.isDebug()) {
+            DEBUG_LOGGER.info(
+                "[LastQueryExecutor] Update last cache for path: {} with timestamp: {}",
+                seriesPaths,
+                resultContainer.get(i).right.getTimestamp());
           }
+        } else {
+          resultContainer.get(i).left = true;
+          cacheAccessors.get(i).markEmpty();
```

Comment:
Please add some comments to explain the whole process. It helps other people to understand the code.